pkg/views/server/configure.go

```diff
@@ -113,23 +135,75 @@ func ConfigurationForm(config *apiclient.ServerConfig, containerRegistries []api
 		huh.NewGroup(
 			huh.NewInput().
 				Title("Frps Domain").
-				Value(config.Frps.Domain),
+				Value(m.config.Frps.Domain),
 			huh.NewInput().
 				Title("Frps Port").
 				Value(&frpsPortView).
-				Validate(createPortValidator(config, &frpsPortView, config.Frps.Port)),
+				Validate(createPortValidator(m.config, &frpsPortView, m.config.Frps.Port)),
 			huh.NewInput().
 				Title("Frps Protocol").
-				Value(config.Frps.Protocol),
+				Value(m.config.Frps.Protocol),
 		),
 	).WithTheme(views.GetCustomTheme())
+}
+
+func (m Model) Init() tea.Cmd {
+	return m.form.Init()
+}
+
+func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+s":
```

Comment:
We should definitely run validation on the current field before quitting and saving.

![Screenshot 2024-07-30 at 12 42 57](https://github.com/user-attachments/assets/4d9ac6cd-8a04-43d3-8be1-6909848753ae)

I managed to save this configuration which is not valid (the dir input is empty). Before I saved, I pressed enter and the validation warned me, then I submitted the form with `ctrl+s` and it went through.

Reply:
Please don't mark conversations as resolved yourself. The reviewer will do that once they confirm the resolution.


https://github.com/user-attachments/assets/24a5e1e4-1474-42ad-a7d9-6cf885fe81ea

As you can see, I again managed to save the form in an invalid state. Your fix works only if the errors are truly active, but if they are not, you can still save the form.

Reply:
Pressing Enter validates or saves the current field value. If there are errors, they are displayed on the TUI. We can use the NextField() function to simulate this validation behavior.